ui/app/pages/unlock-page/tests/unlock-page.test.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import assert from 'assert'
+import sinon from 'sinon'
+import configureMockStore from 'redux-mock-store'
+import { mountWithRouter, stubComponent } from '../../../../../test/lib/render-helpers'
+import UnlockPage from '../index'
+import Mascot from '../../../components/ui/mascot'
+
+describe('Unlock Page', () => {
+  let wrapper
+
+  const props = {
+    history: {
+      push: sinon.spy(),
+    },
+    isUnlocked: false,
+    onImport: sinon.spy(),
+    onRestore: sinon.spy(),
+    onSubmit: sinon.spy(),
+    forceUpdateMetamaskState: sinon.spy(),
+    showOptInModal: sinon.spy(),
+  }
+
+  const mockStore = {
+    metamask: {},
+  }
+
+  const store = configureMockStore()(mockStore)
+
+  stubComponent(Mascot)
```

Comment:
The Mascot is initialized with the `unlock.component`, https://github.com/MetaMask/metamask-extension/blob/7b730ebc472a86c8e9b6f12f8136b20131fed544/ui/app/pages/unlock-page/unlock-page.component.js#L144-L148

The Mascot Component requires a DOM, one that append to and animates with the mouse and key presses. It seems that the enzyme doesn't know how to handle these. Which lead me to stub the react component and return `null` for it to prevent those animations while testing the `unlock.component` in isolation.

Reply:
Ahhh I see, that makes sense. So this is just a means of stubbing it out.

I think this would be better accomplished using something like `proxyquire` to return a stub in place of that component - as this approach of stubbing the Mascot class itself will mutate that class globally for the duration of this Node.js session. This would break any tests on Mascot itself, for example, though it seems in this case it isn't used anywhere else, which is why it's working.

Reply:
I played around with `proxyquire` but could get it to play nicely with the router stuff. Will look into it more, and get more familiar with it since some of the component tests are stubbed with it.

Reply:
After looking into this a bit more, I think you're right that `proxyquire` doesn't handle this case well. `proxyquire` is only meant to stub out dependencies of the module you're importing directly, whereas this is one level deeper than that (as `Mascot` is being imported by the component, which itself is being imported by the container which is what's being imported here).

Though... looking at these tests, they are only really testing the component. One solution would be to import it directly, and test that instead. I managed to get that to work here: https://github.com/Gudahtt/metamask-extension/commit/27711c7347f5cee9f30785850a4b8f5278f1f570

Though another solution might be to just fix Mascot, and stop mutating the DOM. React makes it quite easy to achieve the same ends without doing that. I'll give that a try.

Reply:
The Mascot has been updated in #7893. I've confirmed that once that is merged, the component stubbing done here isn't required anymore.